### PR TITLE
Add rustfmt config file and run over workspace

### PR DESCRIPTION
**Pull request by DevinR528**
Not sure how you want to configure the `rustfmt.toml` but here are the [options](https://rust-lang.github.io/rustfmt/). Unfortunatly this is still nightly only :disappointed: but we could add it to nightly CI. Anyway I was going to submit the PR, figure out the config then delete the `rustfmt.toml` file but keeping all the changes?

### CODE REVIEW

**Comment by DevinR528** [comment]
Just so its easy to see here is what I had
```toml
comment_width = 100
wrap_comments = true
imports_granularity = "Crate"
max_width = 100
group_imports = "StdExternalCrate"

```

File: rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+comment_width = 100
```
Comment (poljar) [inline rustfmt.toml:1]: I think I prefer 80 for this. My screen is ~170 cols wide, so when I split the window to view two files at once I can see the whole doc comment anyways. Any objections to putting this to 80?

Comment (DevinR528) [reply rustfmt.toml:1]: No I'm with you on the skinny width! I'll see how it goes. Do you want me to make the max_width 80 as well or just comments?

Comment (DevinR528) [reply rustfmt.toml:1]: `max_width = 90 and comment_width = 80` looks pretty good.

**Comment by jplatte** [comment]
I would like to advocate for `use_small_heuristics = "Max"` in addition, it produces much nicer output than the default in some cases :)

File: rustfmt.toml (hunk 1 above)
Comment (poljar) [reply rustfmt.toml]: Just comments I would say, most people leave the default for Rust, which is 100?

**Comment by poljar** [comment]
> I would like to advocate for `use_small_heuristics = "Max"` in addition, it produces much nicer output than the default in some cases :)

Much nicer in what way?

**Comment by jplatte** [comment]
More compact. For example (from running `rustfmt` with that option on this repo):

```diff
@@ -1311,9 +1289,7 @@ impl Client {
             let request = FilterUploadRequest::new(&user_id, definition);
             let response = self.send(request, None).await?;
 
-            self.base_client
-                .receive_filter_upload(filter_name, &response)
-                .await?;
+            self.base_client.receive_filter_upload(filter_name, &response).await?;
 
             Ok(response.filter_id)
         }
```

```diff
@@ -2205,11 +2159,7 @@ impl Client {
     #[cfg(feature = "encryption")]
     #[cfg_attr(feature = "docs", doc(cfg(encryption)))]
     pub async fn bootstrap_cross_signing(&self, auth_data: Option<AuthData<'_>>) -> Result<()> {
-        let olm = self
-            .base_client
-            .olm_machine()
-            .await
-            .ok_or(Error::AuthenticationRequired)?;
+        let olm = self.base_client.olm_machine().await.ok_or(Error::AuthenticationRequired)?;
 
         let (request, signature_request) = olm.bootstrap_cross_signing(false).await?;
```

```diff
@@ -2261,10 +2211,7 @@ impl Client {
     ) -> StdResult<UserDevices, CryptoStoreError> {
         let devices = self.base_client.get_user_devices(user_id).await?;
 
-        Ok(UserDevices {
-            inner: devices,
-            client: self.clone(),
-        })
+        Ok(UserDevices { inner: devices, client: self.clone() })
     }
 
     /// Export E2EE keys that match the given predicate encrypting them with the
```

**Comment by jplatte** [comment]
Though be warned, the overall diff from enabling that option is `65 files changed, 1301 insertions(+), 3480 deletions(-)` 😅

**Comment by poljar** [comment]
> Though be warned, the overall diff from enabling that option is `65 files changed, 1301 insertions(+), 3480 deletions(-)` sweat_smile

Nice, that's going to bump our coverage since tarpaulin often times (used to?) has problems with such lines.

**Comment by DevinR528** [comment]
Do we want to change CI to run on nightly and leave the `rustfmt.toml`? Or should I remove the file?

**Comment by jplatte** [comment]
Rename it to `.rustfmt.toml` :P

**Comment by poljar** [comment]
We can run the the "Check style" action on nightly.

**Comment by DevinR528** [comment]
No idea why running `cargo fmt` a third time changed things, although I did `rustup update` in between so maybe that's it?

**Comment by poljar** [comment]
Could you please rerun cargo fmt to get rid of the merge conflicts?

**Comment by johannescpk** [comment]
Just wanted to leave this here for consideration: https://dev.to/alexandersandberg/why-we-should-default-to-tabs-instead-of-spaces-for-an-accessible-first-environment-101f

**Pull request merged**